src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1234,15 +1234,6 @@ private async Task<TokenValidationResult> ValidateTokenAsync(JsonWebToken jsonWe
                 // we want to make sure that the clause for SecurityTokenUnableToValidateException is hit so that the ValidationFailure is checked
                 else if (TokenUtilities.IsRecoverableException(tokenValidationResult.Exception))
                 {
-                    if (TokenUtilities.IsRecoverableConfiguration(validationParameters, currentConfiguration, out currentConfiguration))
```

Comment:
I think the logic for setting LastKnownGood (line 1288) should be just
if (currentConfiguration != null)

Then the logic here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/013a42271866c02574fff4c74cb9ca7d08fc090e/src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs#L75

Should be modified so that when LKG is set, the timer starts.
This means that LKG will be used for only the number of hours AFTER it was last set.
The current logic could result in LKG being valid longer.
Not a big deal, but it's non deterministic.

Reply:
updated the logic